examples/rotating-voronoi/src/index.ts

```diff
@@ -0,0 +1,222 @@
+const center = [width / 2, height / 2];
+
+const rndInt = (min: number, max: number) =>
+    Math.trunc(min + Math.random() * (max - min));
+
+const startingCircles: Array<[number, number, boolean]> = [
+    [radius / 1, rndInt(4, 20), true],
+    [radius / 2, rndInt(4, 20), false],
+    [radius / 4, rndInt(4, 20), true],
+    [radius / 8, rndInt(4, 20), false]
+];
+
+const pointsInCircle = (
+    _center: Vec,
+    _radius: number,
+    _num: number,
+    _angle: number
+) => [
+    ...iterator1(
+        map((index) =>
+            cartesian2([], [_radius, index * TAU + _angle], _center)
+        ),
+        normRange(_num, false)
+    )
+];
+
+scaleStream.next(1);
+frameStreamConditional.next(0);
+animationStream.next(false);
+const startKeyEvent: KeyboardEvent = document.createEvent("KeyboardEvent");
+startKeyEvent.initEvent("keyup");
+keyStreamConditional.next(startKeyEvent);
+
+const app = (...args: any) => (state: AppState) => appRender(state);
+mainStream.transform(
+    map(app(scaleStream, animationStream, frameStream, keyStream)),
+    updateDOM()
+);
+
+function computeVoronoi(state: AppState) {
+    const delta = state.frameValue / 100;
+    const doSave = state.keyValue === "s";
+
+    const opts: CubicOpts = {
+        breakPoints: false,
+        uniform: false,
+        scale: state.scaleValue
+    };
+
+    const startPoints = [
+        ...iterator(
+            comp(
+                map(([rad, density, clockwise]) =>
+                    pointsInCircle(
+                        center,
+                        rad,
+                        density,
+                        clockwise ? delta : PI - delta
+                    )
+                ),
+                mapcat((x) => x)
+            ),
+            startingCircles
+        )
+    ];
+
+    const mesh = new DVMesh();
+    mesh.addKeys(startPoints, 0.01);
+    const bounds = [[0, 0], [width, 0], [width, height], [0, height]];
+    const cells = mesh.voronoi(bounds);
+
+    const voronoi = [
+        rect([width, height], { fill: "black" }),
+
+        ...cells.map((cell) => {
+            const cell0 = new Polygon(simplify(cell, 0.01, true));
+
+            const cubics = asCubic(cell0, opts);
+            const paths = pathFromCubics(cubics);
+
+            return group({}, [
+                withAttribs(paths, { fill: "white", "stroke-width": 1 })
+            ]);
+        }),
+        points(doSave ? [] : startPoints, {
+            size: 4,
+            shape: "circle",
+            fill: "gray"
+        })
+    ];
+
+    if (doSave) {
+        const svg = convertTree(
+            svgDoc(
+                {
+                    width,
+                    height,
+                    viewBox: `0 0 ${width} ${height}`,
+                    "stroke-width": 0.25
+                },
+                ...voronoi
+            )
+        );
+        const finalSvg = serialize(svg);
+        download(`${new Date().getTime()}-voronoi.svg`, finalSvg);
```

Comment:
As mentioned earlier, use `asSvg()` from `geom` package to simplify. This will also auto-compute the `viewBox` attrib. In general `width` and `height` are only needed if there's no viewbox and/or the viewbox size is different than width/height (not the case here):

```ts
const svg = asSvg(
    svgDoc(
        {
            // width,
            // height,
            // viewBox: `0 0 ${width} ${height}`,
            "stroke-width": 0.25
        },
        ...voronoi
    )
);
download(`${new Date().getTime()}-voronoi.svg`, svg)
```